src/atoms/affine/stack.jl

```diff
@@ -132,8 +132,15 @@ function hvcat(rows::Tuple{Vararg{Int}}, args::AbstractExprOrValue...)
     return vcat(rs...)
 end
 
+_isexpr(x::Union{T,Type{T}}) where {T<:AbstractExpr} = true
+_isexpr(x) = false
 
-Base.vect(args::T...) where {T<:AbstractExpr} = transpose(HcatAtom(map(transpose, args)...))
 Base.vect(args::AbstractExpr...) = transpose(HcatAtom(map(transpose, args)...))
-Base.vect(args::AbstractExprOrValue...) = transpose(HcatAtom(map(arg -> transpose(convert(AbstractExpr, arg)), args)...))
-Base.vect(args::Value...) = collect(args)
+
+function Base.vect(args::AbstractExprOrValue...)
+    if any(_isexpr, args)
+        transpose(HcatAtom(map(arg -> transpose(convert(AbstractExpr, arg)), args)...))
+    else
+        invoke(Base.vect, Tuple{Vararg{Any}}, args...)
```

Comment:
Does this break type inference for `Base.vect`?

Reply:
Luckily it does not seem to. Using the example in the linked issue:
```julia
julia> @code_warntype [1, 1im]
Body::Array{Complex{Int64},1}
141 1 ── %1  = (Base.getfield)(args, 1, true)::Int64                                                                                  │╻╷╷╷╷ any
    └───       goto #12 if not true                                                                                                   ││┃     _any
    2 ┄─ %3  = φ (#1 => %1, #11 => %19)::Union{Complex{Int64}, Int64}                                                                 │││   
    │    %4  = φ (#1 => 2, #11 => %20)::Int64                                                                                         │││   
    │    %5  = (isa)(%3, Complex{Int64})::Bool                                                                                        │││   
    └───       goto #4 if not %5                                                                                                      │││   
    3 ──       goto #7                                                                                                                │││   
    4 ── %8  = (isa)(%3, Int64)::Bool                                                                                                 │││   
    └───       goto #6 if not %8                                                                                                      │││   
    5 ──       goto #7                                                                                                                │││   
    6 ──       (Core.throw)(ErrorException("fatal error in type inference (type bound)"))                                             │││   
    └───       $(Expr(:unreachable))                                                                                                  │││   
    7 ┄─ %13 = (Base.slt_int)(2, %4)::Bool                                                                                            │││╻╷    iterate
    └───       goto #9 if not %13                                                                                                     ││││  
    8 ──       goto #10                                                                                                               ││││  
    9 ── %16 = (Base.getfield)(args, %4, true)::Union{Complex{Int64}, Int64}                                                          ││││╻     getindex
    │    %17 = (Base.add_int)(%4, 1)::Int64                                                                                           ││││╻     +
    └───       goto #10                                                                                                               ││││  
    10 ┄ %19 = φ (#9 => %16)::Union{Complex{Int64}, Int64}                                                                            │││   
    │    %20 = φ (#9 => %17)::Int64                                                                                                   │││   
    │    %21 = φ (#8 => true, #9 => false)::Bool                                                                                      │││   
    │    %22 = (Base.not_int)(%21)::Bool                                                                                              │││   
    └───       goto #12 if not %22                                                                                                    │││   
    11 ─       goto #2                                                                                                                │││   
    12 ─       goto #13                                                                                                               │││   
    13 ─       goto #14                                                                                                               ││    
    14 ─       goto #16 if not false                                                                                                  │     
    15 ─       nothing                                                                                                                │     
144 16 ─ %29 = Base.vect::Core.Compiler.Const(Base.vect, false)                                                                       │     
    │    %30 = (getfield)(args, 1)::Int64                                                                                             │     
    │    %31 = (getfield)(args, 2)::Complex{Int64}                                                                                    │     
    │    %32 = invoke %29(%30::Int64, %31::Vararg{Any,N} where N)::Array{Complex{Int64},1}                                            │     
    └───       return %32                                                                                                             │     
```